plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py

```diff
@@ -14,20 +14,33 @@
     StructuredDatasetTransformerEngine,
 )
 
-pd = lazy_module("pandas")
-pl = lazy_module("polars")
-fsspec_utils = lazy_module("fsspec.utils")
+if typing.TYPE_CHECKING:
+    import polars as pl
+    import fsspec.utils as fsspec_utils
+else:
+    pl = lazy_module("polars")
+    fsspec_utils = lazy_module("fsspec.utils")
 
 
+############################
+# Polars DataFrame classes #
+############################
+
 class PolarsDataFrameRenderer:
     """
     The Polars DataFrame summary statistics are rendered as an HTML table.
     """
 
-    def to_html(self, df: pl.DataFrame) -> str:
-        assert isinstance(df, pl.DataFrame)
-        describe_df = df.describe()
-        return pd.DataFrame(describe_df.transpose(), columns=describe_df.columns).to_html(index=False)
+    def to_html(self, df: typing.Union[pl.DataFrame, pl.LazyFrame]) -> str:
+        assert isinstance(df, (pl.DataFrame, pl.LazyFrame))
+        if isinstance(df, pl.LazyFrame):
+            describe_df = df.collect().describe()
```

Comment:
`describe` does not need to be collected first with a lazy dataframe:

```python
import polars as pl

df = pl.DataFrame({"A": [1, 2, 3], "b": [4, 5, 6]}).lazy()
assert isinstance(df, pl.LazyFrame)

df.describe()
```

Reply:
it does for polars <= 0.19. Unless we place a lower bound on the polars version supported by this plugin.

Reply:
Can we try `describe` first and fallback to `collect` if it does not work?

```python
try:
    describe_df = df.describe()
except AttributeError:
    # LazyFrames in polars <= 0.19 does not support `describe` 
    describe_df = df.collect().describe()
```